convert_hf_to_gguf.py

```diff
@@ -1514,6 +1514,35 @@ def set_gguf_parameters(self):
         if self.hparams.get("vocab_size", 32000) == 49152:
             self.gguf_writer.add_add_bos_token(False)
 
+        if rope_scaling := self.find_hparam(["rope_scaling"], optional=True):
+            if rope_scaling.get("rope_type", '').lower() == "llama3":
```

Comment:
I think it's best if you merge this into the rope scaling above (line 1501), so that all the rope operations are in one place.